ServerCore/Pages/Threads/PuzzleThreads.cshtml

```diff
@@ -138,6 +141,12 @@
                 <td>
                     <a asp-page="/Threads/PuzzleThread" asp-route-puzzleId="@item.PuzzleID" asp-route-teamId="@item.TeamID" asp-route-playerId="@item.PlayerID">@item.Subject</a>
                 </td>
+                <td>
+                    @if (@item.PuzzleID != null)
+                    {
+                        @Model.AuthorsForPuzzleID[item.PuzzleID ?? 0]
```

Comment:
Don't need to double-protect here; you know it's not null at this point.

Reply:
I thought the same, but even with the null check in the if, there's an error in converting from int? to int for PuzzleID.  `@Model.AuthorsForPuzzleID[(int)item.PuzzleID]` would also solve the error.  Is casting preferable when we know it's safe?  Removing the if could also work, but defaulting to 0 could technically give incorrect data if something goes wrong in creating the dictionary.

Reply:
item.PuzzleID.Value would be fine and we use that all over (and HasValue for the null check if you feel like it, though we're less consistent with that)

Reply:
Ooh even better, thanks!